### ISSUE TITLE
Modal not getting dismissed.

### ISSUE THREAD

**Issue by angular2fan** [dougludlow/ng2-bs3-modal#174]
Modal not getting dismissed.

tried debugging the code and in the below function in modal-instance.ts file, promise is not defined is the message I am seeing.

```
 private hide(): Promise<ModalResult> {
        if (this.$modal && this.visible) {
            let promise = toPromise(this.hidden);
            this.$modal.modal('hide');
            return promise;
        }
        return Promise.resolve(this.result);
}
```

How am i calling the code??
//modal is a differnt modal window in which I have a form and I do something in that form and then make a http call. when the http call is in flight, I want to display the please wait in a modal window(loadingscreenmodal).
 
```
this.modal.dismiss();
this.loadingScreenModal.open('sm');
           
            this.servicecall(param1,
                tparam2,
                param3,
                param4)
                .then((response) => {
                    
                    this.loadingScreenModal.dismiss();
}
```

The code is executing the dismiss statement, but is failing at the above line of code in modal-instance.ts file.

Any Help or any clarification on what could be going on?

Thanks.

**Comment by wartab** [comment]
Where are you seeing this message? Upon compilation at runtime? In any case, this seems more like a dependency configuration issue on your side.

**Comment by angular2fan** [comment]
hmmm, my application is not making use of AOT yet and is using systemjs.
I am debugging the application to see why the modal is not getting dismissed and then i am seeing this message. Thanks

**Comment by angular2fan** [comment]
just an update. I spent some more time debugging and I noticed that the dismiss behavior in this case is little strange because it is working sometimes and failing sometimes but whenever it is failing I saw that Line 56 in modal-instance.ts [let promise = toPromise(this.hidden);] is not even getting executed in my case. Why is the toPromise function not even getting called? I am seeing this behavior across all three browsers (IE, FF and Chrome)

Any insight?? Thanks in advance
